superset/dashboards/api.py

```diff
@@ -589,3 +596,48 @@ def thumbnail(
         return Response(
             FileWrapper(screenshot), mimetype="image/png", direct_passthrough=True
         )
+
+    @expose("/favorite_status/", methods=["GET"])
+    @protect()
+    @safe
+    @statsd_metrics
+    @rison(get_fav_star_ids_schema)
+    def favorite_status(self, **kwargs: Any) -> Response:
+        """Favorite Stars for Dashboards
+        ---
+        get:
+          description: >-
+            Check favorited dashboards for current user
+          parameters:
+          - in: query
+            name: q
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/get_fav_star_ids_schema'
+          responses:
+            200:
+              description:
+              content:
+                application/json:
+                  schema:
+                    $ref: "#/components/schemas/GetFavStarIdsSchema"
+            400:
+              $ref: '#/components/responses/400'
+            401:
+              $ref: '#/components/responses/401'
+            404:
+              $ref: '#/components/responses/404'
+            500:
+              $ref: '#/components/responses/500'
+        """
+        requested_ids = kwargs["rison"]
+        dashboards = DashboardDAO.find_by_ids(requested_ids)
```

Comment:
Wouldn't this load all charts from db? Would it be OK if we just pass a list of chart ids to `DashboardDAO.favorited_ids` so we don't have to run the additional query to get all the charts?

Reply:
Unfortunately we need to run this additional query. The find_by_ids method has business logic to ensure the user has access to the requested dashboard by applying the dashboard filter (https://github.com/preset-io/incubator-superset/blob/cb866b30c0a7029ff558297b1f15c7328f64c500/superset/dashboards/filters.py#L59). We could try to fetch the favorite_ids with the dashboard filter in a subquery, but the performance gains wouldn't be significant. We're also doing something similar for bulk delete already (https://github.com/preset-io/incubator-superset/blob/cb866b30c0a7029ff558297b1f15c7328f64c500/superset/dashboards/commands/bulk_delete.py#L54).